esteid/digidocservice/service.py

```diff
@@ -1,37 +1,40 @@
 # -*- coding: utf-8 -*-
-# Needs suds_jurko
 import base64
 import binascii
 import logging
 import os
 
 from django.utils.encoding import force_text, force_bytes
-from suds import WebFault
-from suds.client import Client
-from suds.plugin import MessagePlugin
+
+from zeep import Client, Transport
+from zeep.cache import SqliteCache
+from zeep.exceptions import Fault
+from zeep.xsd import SkipValue
 
 from .containers import BdocContainer
+from .util import get_bool, get_optional_bool
 
 
 class DigiDocException(Exception):
     """ Unknown errors
     """
 
-    def __init__(self, command, params, *args, **kwargs):
+    def __init__(self, command, params, *args):
         self.command = command
         self.params = params
 
-        super().__init__(*args, **kwargs)
+        super(DigiDocException, self).__init__(*args)
```

Comment:
Any reason for prefering super(ClassName, self) over super() ?

Reply:
Py2 support

Reply:
Test output on python 2.7.12 without this change:

```
py.test
======================================================================================================================== test session starts ========================================================================================================================
platform linux2 -- Python 2.7.12, pytest-3.0.4, py-1.4.34, pluggy-0.4.0
django settings: test_settings (from ini file)
rootdir: /home/jyrno42/Projektid/Progremine/thorgate/django-esteid, inifile: pytest.ini
plugins: django-2.9.1, cov-2.4.0
collected 9 items 

tests/tests.py ...F.....

============================================================================================================================= FAILURES ==============================================================================================================================
_______________________________________________________________________________________________________ TestMobileAuthenticate.test_mobileauthenticate_error ________________________________________________________________________________________________________

self = <tests.tests.TestMobileAuthenticate testMethod=test_mobileauthenticate_error>

    def test_mobileauthenticate_error(self):
        # From https://www.id.ee/?id=36381
        failing_numbers = [
            ('14212128027', 'EE', '+37200009', 302),
            ('38002240211', 'EE', '+37200001', 303),
        ]
    
        for id_code, country, phone_nr, error_code in failing_numbers:
            service = get_digidoc_service()
    
            with pytest.raises(DigiDocError) as e:
>               service.mobile_authenticate(id_code=id_code, country=country, phone_nr=phone_nr)

tests/tests.py:119: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
esteid/digidocservice/service.py:163: in mobile_authenticate
    }, no_raise=True)
esteid/digidocservice/service.py:390: in __invoke
    "Server result [/%s - %s]: %s" % (command, error_code, known_fault))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = DigiDocError(), error_code = 302, known_fault = 'Sertifikaat ei kehti (OCSP: TAGASI V\xc3\x95ETUD).', args = ('Server result [/MobileAuthenticate - 302]: Sertifikaat ei kehti (OCSP: TAGASI V\xc3\x95ETUD).',)

    def __init__(self, error_code, known_fault, *args):
        self.error_code = error_code
        self.known_fault = known_fault
    
>       super().__init__(*args)
E       TypeError: super() takes at least 1 argument (0 given)

esteid/digidocservice/service.py:37: TypeError
================================================================================================================ 1 failed, 8 passed in 34.45 seconds ================================================================================================================
Makefile:28: recipe for target 'test' failed
make: *** [test] Error 1
ERROR: InvocationError: '/usr/bin/make test'
______________________________________________________________________________________________________________________________ summary ______________________________________________________________________________________________________________________________
ERROR:   py27-django18: commands failed

```

Reply:
Yeah absolutely, didn't know we were supporting py2, got it.